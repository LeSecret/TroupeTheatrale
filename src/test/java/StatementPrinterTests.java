import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import static org.approvaltests.Approvals.verify;

public class StatementPrinterTests {

    @Test
    void exampleStatement() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet",  new Tragedy("Hamlet"));
        plays.put("as-like",  new Comedy("As You Like It"));
        plays.put("othello",  new Tragedy("Othello"));

        Invoice invoice = new Invoice(new Customer("BigCo", "1", 103), List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.toText(invoice, plays);

        verify(result);
    }

    @Test
    void exampleStatementCustomer() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet",  new Tragedy("Hamlet"));
        plays.put("as-like",  new Comedy("As You Like It"));
        plays.put("othello",  new Tragedy("Othello"));

        Invoice invoice = new Invoice(new Customer("BigCo", "1", 150), List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.toText(invoice, plays);

        verify(result);
    }

    @Test
    void exampleStatementHTML() {
        Map<String, Play> plays = Map.of(
                "hamlet",  new Tragedy("Hamlet"),
                "as-like", new Comedy("As You Like It"),
                "othello", new Tragedy("Othello"));

        Invoice invoice = new Invoice(new Customer("BigCo", "1", 0), List.of(
                new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        StatementPrinter statementPrinter = new StatementPrinter();
        var result = statementPrinter.toHTML(invoice, plays);

        verify(result);
    }

    /*@Test
    void statementWithNewPlayTypes() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("henry-v",  new Play("Henry V", "history"));
        plays.put("as-like",  new Play("As You Like It", "pastoral"));

        Invoice invoice = new Invoice("BigCo", List.of(
              verify(result);
          new Performance("henry-v", 53),
                new Performance("as-like", 55)));

        StatementPrinter statementPrinter = new StatementPrinter();
        Assertions.assertThrows(Error.class, () -> {
            statementPrinter.print(invoice, plays);
        });
    }*/
}
