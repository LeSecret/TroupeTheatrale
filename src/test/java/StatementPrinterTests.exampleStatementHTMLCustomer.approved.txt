<!DOCTYPE html> 
	<html> 
		<body> 
			<h1>Invoice</h1> 
			<p><b>Client :</b> BigCo</h2> 
			<table style="width: 400px" border="1px solid black"> 
				<tr> 
					 <th>Piece</th> 
					 <th>Seats sold</th> 
					 <th>Price</th> 
				</tr> 
				<tr> 
					 <td>Hamlet</td> 
					 <td>55</td> 
					 <td>$650.00</td> 
				</tr> 
				<tr> 
					 <td>As You Like It</td> 
					 <td>35</td> 
					 <td>$580.00</td> 
				</tr> 
				<tr> 
					 <td>Othello</td> 
					 <td>40</td> 
					 <td>$500.00</td> 
				</tr> 
				<tr> 
					 <td colspan="2" style="text-align: right;"><b>Total owed:<b></td> 
					 <td>$1,715.00</td> 
				</tr> 
				<tr> 
					 <td colspan="2" style="text-align: right;"><b>Fidelity points earned:<b></td> 
					 <td>0</td> 
				</tr> 
			</table> 
			<p>You have been deducted of 150 credits and 15$</p> 
			<p>Payment is required under 30 days. We can break your knees if you don't do so.</p> 
		</body> 
	</html> 